### PR TITLE
Fix 6279 wrong validation of projectid

### PR DESCRIPTION
**Pull request by nick2432**
## What does this PR do?

update the helper text 

Fixes https://github.com/appwrite/appwrite/issues/6279

## Test Plan

I update the CustomId component when name is project, it will use a new InputProjectId rather than InputId.

I updated the unit tests, and I verified that they all passed successfully.

before
before

after
after
Screenshot from 2023-10-16 10-13-59
d58c280610)

## Related PRs and Issues

- https://github.com/appwrite/appwrite/issues/6279

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

(Write your answer here.)

### CODE REVIEW

File: src/lib/components/customId.svelte [hunk 1]
```diff
@@ -29,7 +29,11 @@
     </svelte:fragment>
     <svelte:fragment slot="content">
         <div class="form">
-            <InputId bind:value={id} {autofocus} />
+            {#if name === 'Project'}
```
Comment (TorstenDittmann) [inline src/lib/components/customId.svelte:32]: I think we should add a new exported variable called

`export let isProject = false;`

Becuase we will introduce i18n at some point and this doesnt scale 👍🏻

**Review by TorstenDittmann — CHANGES_REQUESTED** [review]

File: src/lib/components/customId.svelte (hunk 1 above)
Comment (nick2432) [reply src/lib/components/customId.svelte]: Thank you for letting me know, and I have made the changes

File: package.json [hunk 2]
```diff
@@ -55,7 +55,7 @@
         "@typescript-eslint/eslint-plugin": "^6.5.0",
         "@typescript-eslint/parser": "^6.5.0",
         "@vitest/ui": "^0.29.7",
-        "eslint": "^8.48.0",
+        "eslint": "^8.51.0",
```
[the remark this answers is not in the document]
Comment (nick2432) [reply package.json:58]: There's no need for it.

File: src/lib/components/customId.svelte [hunk 3]
```diff
@@ -29,7 +29,11 @@
     </svelte:fragment>
     <svelte:fragment slot="content">
         <div class="form">
-            <InputId bind:value={id} {autofocus} />
+            {#if isProject === true}
```
Comment (TorstenDittmann) [inline src/lib/components/customId.svelte:32]:
```suggestion
            {#if isProject}
```

File: src/routes/console/onboarding/+page.svelte [hunk 4]
```diff
@@ -64,7 +64,7 @@
                         </Pill>
                     </div>
                 {:else}
-                    <CustomId bind:show={showCustomId} name="Project" bind:id />
+                    <CustomId bind:show={showCustomId} name="Project" isProject={true} bind:id />
```
Comment (TorstenDittmann) [inline src/routes/console/onboarding/+page.svelte:67]:
```suggestion
                    <CustomId bind:show={showCustomId} name="Project" isProject bind:id />
```

File: src/routes/console/organization-[organization]/createProject.svelte [hunk 5]
```diff
@@ -56,7 +56,7 @@
                 </Pill>
             </div>
         {:else}
-            <CustomId bind:show={showCustomId} name="Project" bind:id />
+            <CustomId bind:show={showCustomId} name="Project" isProject={true} bind:id />
```
Comment (TorstenDittmann) [inline src/routes/console/organization-[organization]/createProject.svelte:59]:
```suggestion
            <CustomId bind:show={showCustomId} name="Project" isProject bind:id />
```

**Review by TorstenDittmann — COMMENTED** [review]
lets use some shorter syntax and we are good imo 👍🏻

File: package-lock.json [hunk 6]
```diff
@@ -43,7 +43,7 @@
                 "@typescript-eslint/eslint-plugin": "^6.5.0",
                 "@typescript-eslint/parser": "^6.5.0",
                 "@vitest/ui": "^0.29.7",
-                "eslint": "^8.48.0",
+                "eslint": "^8.51.0",
```
Comment (stnguyen90) [inline package-lock.json:46]: Please remove the package-lock.json change from the PR too.

**Review by stnguyen90 — CHANGES_REQUESTED** [review]
Tested and the validation works well:

image

The only minor recommendation is to remove the package-lock.json change.

**Review by stnguyen90 — APPROVED** [review]

**Review by TorstenDittmann — APPROVED** [review]

**Pull request merged**